shards/rust/src/util.rs

```diff
@@ -89,6 +90,83 @@ pub fn collect_required_variables(
   Ok(())
 }
 
+extern "C" {
+  fn shards_collect_required_variables_typed(
+    data: *const SHInstanceData,
+    out: *mut SHExposedTypesInfo,
```

Comment:
This should go through SHCore

Reply:
does it matter? it's kinda very internal and rust is basically mandatory now due to cli etc anyway

Reply:
collectRequiredVariables

Reply:
I'm slightly confused as to why bg3 shards don't fail to build though